src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -1261,6 +1314,20 @@ class query extends viewModelBase {
             cache.total(cache.items.size);
         });
     }
+
+    private onIncludesRevisionsLoaded(includesRevisionsFromQuery: Array<any>): void {
+        includesRevisionsFromQuery.forEach(x => {
```

Comment:
I think includesRevisionsCache doesn't have to be observable. 

Here instead of pushing to array map items and push them all in single call - it willbe faster.

```
const mappedItems = includesRevisionsFromQuery.map(x => ...);

this.includesRevisionsCache.items.push(...mappedItems)
```

Reply:
1. done - mapped items first before push
2. cannot really make it not observable, otherwise, the bottom tab still shows when making a different query that does not involve included revisions